Source/CesiumRuntime/Public/CesiumOriginShiftComponent.h

```diff
@@ -0,0 +1,172 @@
+// Copyright 2020-2023 CesiumGS, Inc. and Contributors
+
+#pragma once
+
+#include "Components/ActorComponent.h"
+#include "CoreMinimal.h"
+#include "CesiumOriginShiftComponent.generated.h"
+
+class UCesiumGlobeAnchorComponent;
+
+/**
+ * Indicates how to shift the origin as the Actor to which a
+ * CesiumOriginShiftComponent is attached moves.
+ */
+UENUM(BlueprintType)
+enum class ECesiumOriginShiftMode : uint8 {
+  /**
+   * This component is disabled and will have no effect.
+   */
+  Disabled,
+
+  /**
+   * The origin of the CesiumGeoreference will be changed when the Actor enters
+   * a new sub-level, but it will otherwise not be modified as the Actor moves.
+   * Any objects that are not anchored to the globe with a
+   * CesiumGlobeAnchorComponent will appear to move when the Actor enters a
+   * sub-level.
+   */
+  SwitchSubLevelsOnly,
+
+  /**
+   * The origin of the CesiumGeoreference will change as the Actor moves in
+   * order to maintain small, precise coordinate values near the Actor, as well
+   * as to keep the globe's local "up" direction aligned with the +Z axis. Any
+   * objects that are not anchored to the globe with a
+   * CesiumGlobeAnchorComponent will appear to move whenever the origin changes.
+   *
+   * When using this mode, all Cesium3DTileset instances as well as any Actors
+   * with a CesiumGlobeAnchorComponent need to be marked Movable, because these
+   * objects _will_ be moved when the origin is shifted.
+   */
+  ChangeCesiumGeoreference,
+
+  /**
+   * Unreal's built-in "World Origin Location" will change as the Actor moves in
+   * order to maintain small, precise coordinate values near the Actor. Because
+   * this is a translation only (no rotation), +Z will not stay aligned with the
+   * globe's local "up" direction. Any objects that are not anchored to the
+   * globe with a CesiumGlobeAnchorComponent will appear to move when the Actor
+   * enters a sub-level, but objects will generally maintain their position
+   * otherwise, as long as they respond correctly to Unreal's ApplyWorldOffset
+   * method.
+   */
+  ChangeWorldOriginLocation
```

Comment:
This happens because a globe anchored Actor's ECEF position is only a function of the CesiumGeoreference origin (not its transform) and the Actor position itself. Changing the World Origin changes the Actor position, so it inadvertently affects the longitude/latitude/height.

It should be possible to modify CesiumGeoreference to make it take the WorldOrigin into account in its ECEF<->Unreal transformations. But it will be a hassle, and I'm not sure it's worth it. I think for now I will just remove this `ChangeWorldOriginLocation` option entirely. We can certainly re-add it in the future if we see value in doing so.